catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CacheQueryFactory.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.cache.solr.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.opengis.filter.Filter;
+
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.solr.SchemaFields;
+
+public class CacheQueryFactory {
+
+    private FilterBuilder builder;
+
+    public CacheQueryFactory(FilterBuilder filterBuilder) {
+        builder = filterBuilder;
+    }
+
+    QueryRequest getQueryRequestWithSourcesFilter(QueryRequest input) {
+        QueryRequest queryWithSources = input;
+        List<Filter> sourceFilters = new ArrayList<>();
+        if (input.getSourceIds() != null) {
+            for (String sourceId : input.getSourceIds()) {
+                sourceFilters.add(builder.attribute(StringUtils.removeEnd(SolrCache.METACARD_SOURCE_NAME,
+                        SchemaFields.TEXT_SUFFIX))
+                        .is()
+                        .equalTo()
+                        .text(sourceId));
+            }
+            QueryImpl sourceQuery = new QueryImpl(builder.allOf(input.getQuery(),
+                    builder.anyOf(sourceFilters)));
```

Comment:
The catalog ui frontend change will be address in https://codice.atlassian.net/browse/DDF-2800